pkg/scheduler/algorithm/predicates/metadata.go

```diff
@@ -52,8 +58,14 @@ type predicateMetadata struct {
 	podBestEffort bool
 	podRequest    *schedulercache.Resource
 	podPorts      []*v1.ContainerPort
-	//key is a pod full name with the anti-affinity rules.
-	matchingAntiAffinityTerms map[string][]matchingPodAntiAffinityTerm
+
+	// A map of antiffinity terms' topology pairs to the pods'
+	// that can potentially match the affinity rules of the pod
+	topologyPairToAntiAffinityPods map[topologyPair][]*v1.Pod
+
+	// Reverse map for topologyPairToAntiAffinityPods to reduce deletion time
+	antiAffinityPodToTopologyPairs map[*v1.Pod][]topologyPair
```

Comment:
Since these two maps must always be kept in sync, you should put them in a struct and add methods to the struct for adding/removing Pods to the maps. This reduces likelihood of introducing bugs in future developments and also makes the code cleaner and easier to read.